File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String ret="";
        int mode = 0;
        for(int idx=0; idx<code.length(); idx++){
            if(code.charAt(idx)=='1'){
                mode = mode == 0 ? 1 : 0;
                continue;
            }
            if(mode==0){
                if(idx%2==0)
                    ret+=code.charAt(idx);
            } 
            else{
                if(idx%2==1)
                ret+=code.charAt(idx);
        }
            
        }
        if(ret.equals(""))
            ret="EMPTY";
        return ret;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    mode=0
    for idx in range(0,len(code)):
        if code[idx]=="1":
            mode=not mode 
            continue
        if mode==0:
            if idx%2==0:
                answer+=code[idx]
        else:
            if idx%2!=0:
                answer+=code[idx]
    if answer=="":
        return "EMPTY"
    return answer